lib/puppet/application/agent.rb

```diff
@@ -354,6 +356,17 @@ def run_command
     end
   end
 
+  def log_config()
+    #skip also config reading and parsing if debug is not enabled
+    return unless Puppet::Util::Log.level == :debug
+
+    config_face = Puppet::Face["config", "0.0.1"]
+    config_face.print(section: "main").each_pair do |k,v|
+      next if k.include?("password") || v.to_s.empty?
+      Puppet.debug("Using setting #{k}=#{v}")
+    end
```

Comment:
Sorry I missed this last time. I'd prefer that the `agent` application didn't know about the `config` face to print the settings. I'd suggest adding another commit to refactoring most of the code in the `config` face's `when_invoked` block to an instance method on the `Puppet::Settings` class, and having the `config` and `agent` applications call that method. The new method should require a section and optionally the name of the setting or default to all. Something like

```ruby
def stringify_settings(section, settings: :all)
```

The `config` app should contain the logic that detects if a section was not specified, and warn, something like:

```ruby
      @default_section = false
      if options[:section] == DEFAULT_SECTION_MARKER
        options[:section] = DEFAULT_SECTION
        @default_section = true
      end

      names = if args.empty? || args == ['all']
                       :all
                      else
                        args
                      end
      if Puppet::Util::Log.sendlevel?(:info)
        warn_default_section(options[:section]) if @default_section
        report_section_and_environment(options[:section], Puppet.settings[:environment])
      end

     to_be_rendered = Puppet.settings.stringify_settings(options[:section], settings: names)
     ...
```

While the `agent` app should always pass section `:agent`:

```ruby
 Puppet.settings.stringify_settings(:agent, settings: :all).each_pair do |k,v|
   ..
end
```